letsencrypt/cli.py

```diff
@@ -1051,6 +1030,51 @@ def _plugins_parsing(helpful, plugins):
 
     helpful.add_plugin_args(plugins)
 
+    # These would normally be a flag within the webroot plugin, but because
+    # they are parsed in conjunction with --domains, they live here for
+    # legibiility. helpful.add_plugin_ags must be called first to add the
+    # "webroot" topic
+    helpful.add("webroot", "-w", "--webroot-path", action=WebrootPathProcessor,
+                help="public_html / webroot path")
+    parse_dict = lambda s: dict(json.loads(s))
+    helpful.add("webroot", "--webroot-map", default={}, type=parse_dict,
```

Comment:
I started working on some code to make `-d` unecessary when `--webroot-map` is present, but the tests turned out to be tricky, so I'm not going to spend more time on it for now.  Will hide --webroot-map instead.